OPEX-1054: Adds health tab documentation. (#8864)
auth0/docs: articles/logs/streams/health.md

---
title: Log Streams Health
description: All Log Streams types let you view your stream health in the health tab.
toc: false
topics:
 - logs
 - streams
 - event-streams
contentType: how-to
---

> # HTTP Event Log Streams

Reviewer: not sure this part is necessary since the explanation of what event streams are is elsewhere.

> # HTTP Event Log Streams

Reviewer 2: The `#` (h1) should match the title: `Log Streams Health`

HTTP Event Log Streams let you export your log events to the server or target of your choice. When Auth0 creates a log entry for your tenant, a copy is automatically sent to a given URL via an HTTP POST request.

If you use Amazon Web Services, Auth0 also offers an [AWS EventBridge integration](/integrations/aws-eventbridge).

> 

Reviewer: Before we dive into the how, maybe give a quick overview of what is log stream health.

# What is Log Stream Health

Log Stream Health is the functionality that allows customers to troubleshoot issues with Log Streams. It allows customers to get the status of the latest delivery attempts of each stream to make it easy to identify problems in the delivery.

# How to use Stream Health

** add some stuff here**

Status types:

| Status | Description |
|---------|-------------|
| Active (with no errors) | This means that the latest log deliveries have been successful |
| Active (with errors) | The log deliveries are active but the stream failed to deliver logs at one or more points |
| Paused | The customer has pausued the stream. |
| Suspended | Our system has disabled the stream because there is an issue and your logs could not be delivered. You may restart the stream by following the directions or returning to list view and clicking the dropdown option "restart" |

## Viewing an Existing Event Stream's Health

1. Log in to the [Auth0 Dashboard](${manage_url}).
2. Navigate to **Logs > Streams**.

> 3. Click on any existing stream listed therein.

Reviewer: ```suggestion
3. Click on a stream
```

> 4. Select the **Health** tab. An explanation of the following potential states is as follows:

Reviewer: ```suggestion
4. Select the **Health** tab

| Status | Description |
|---------|-------------|
| Active (with no errors) | This means that the latest log deliveries have been successful |
| Active (with errors) | The log deliveries are active but the stream failed to deliver logs at one or more points |
| Paused | The customer has pausued the stream. |

> | Suspended | Our system has disabled the stream because there is an issue and your logs could not be delivered. You may restart the stream by following the directions or returning to list view and clicking the dropdown option "restart" |

Reviewer: ```
```

Reviewer 2: Maybe change:

`This means that the latest log deliveries have been successful`

to

`The latest log deliveries have been successful` or `The latest log deliveries were successful`

---
title: Log Streams Health
description: All Log Streams types let you view your stream health in the health tab.
toc: false
topics:
 - logs
 - streams
 - event-streams
contentType: how-to
---

# Viewing an Existing Event Stream's Health

Log Stream Health is the functionality that allows customers to troubleshoot issues with Log Streams. It allows customers to get the status of the latest delivery attempts of each stream to make it easy to identify problems in the delivery.

## Viewing an Existing Event Stream's Health

1. Log in to the [Auth0 Dashboard](${manage_url}).
2. Navigate to **Logs > Streams**.
3. Click on a stream.
4. Select the **Health** tab. 

## Understanding Stream Health States

**add some stuff here**

Status types:

> | Status | Description |

Reviewer 3: I think these should be described in a user friendly way in terms of scenarios. Maybe like what we've been talking about in slack?

https://auth0.slack.com/archives/CEJHNE5QV/p1585588481287000?thread_ts=1585584951.281100&cid=CEJHNE5QV

| Active (with errors) | The log deliveries are active but the stream failed to deliver logs at one or more points. |
| Paused | The customer has paused the stream |
| Suspended | Our system has disabled the stream because there is an issue and your logs could not be delivered. You may restart the stream by following the directions or returning to list view and clicking the dropdown option `restart` |

## Streaming delivery errors, pauses, or suspensions. 

Auth0 events are delivered to your server via a streaming mechanism that sends each event as it is triggered in our system. 

### Errors

Errors are populated in the Logs Health Table when logs could not be delivered with corresponding error codes and counter information. TBD here is how to use counter information to recover logs and the time frame for recovery TBD.

### Pause/Resume Streams

You may pause your stream by using the drop down option menu in the log stream list view and clicking 'pause.' Streams may be resumed by returning to the dropdown and clicking 'resume.'

If a stream is paused, TBD here is how to recover your logs during the pause time frame and the time frame for recovery TBD.

> 

Reviewer 3: We could add a screenshot for pause/resume

### Suspensions

> If your server is unable to receive the stream, we will retry up to fifty times to deliver the event; otherwise, we will suspend the stream. You can restart the stream by following the link in the suspension message or by returning to the list view and using the dropdown menu to click 'restart.' TBD here is how to recover logs during hte suspension state  and and the time frame for recoveryTBD (if all of the above recovery methods are the same, perhaps put it under subheading resolving errors and undelivered logs during paused, suspended, or active states  )

Reviewer 3: screenshots would be helpful here too

> If your server is unable to receive the stream, we will retry up to fifty times to deliver the event; otherwise, we will suspend the stream. You can restart the stream by following the link in the suspension message or by returning to the list view and using the dropdown menu to click 'restart.' TBD here is how to recover logs during hte suspension state  and and the time frame for recoveryTBD (if all of the above recovery methods are the same, perhaps put it under subheading resolving errors and undelivered logs during paused, suspended, or active states  )

Reviewer 3: Let's not make a gaurantee of the number `50`. That's an internal implementation detail we don't want to expose to customers

---
title: Log Streams Health

> description: All Log Streams types let you view your stream health in the health tab.

Reviewer 3: This sentence could be better structured as "View the health for any type of log stream in the health tab"

toc: false
topics:
 - logs
 - streams
 - event-streams
contentType: how-to
---

# Viewing an Existing Event Stream's Health

> Log Stream Health is the functionality that allows customers to troubleshoot issues with Log Streams. It allows customers to get the status of the latest delivery attempts of each stream to make it easy to identify problems in the delivery.

Reviewer: ```suggestion
Log Stream Health allows customers to troubleshoot issues with Log Streams. It doesn it by tracking the status of the latest delivery attempts of each stream to make it easy to identify problems in the delivery.
```

> ## Viewing an Existing Event Stream's Health

Reviewer: ```suggestion
## Viewing the Health of an Existing Log Stream
```

1. Log in to the [Auth0 Dashboard](${manage_url}).
2. Navigate to **Logs > Streams**.
3. Click on a stream.
4. Select the **Health** tab. 

> ## Understanding Stream Health States

Reviewer: ```suggestion
## Understanding Stream Health Status
```

I think is important to be consistent with the terms

**Stream Health has a variety of different states. These status types are described below.**

Status types:

| Status | Description |
|---------|-------------|
| Active (with no errors) | The latest log deliveries were successful |
| Active (with errors) | The log deliveries are active but the stream failed to deliver logs at one or more points. |
| Paused | The customer has paused the stream |

> | Suspended | Our system has disabled the stream because there is an issue and your logs could not be delivered. You may restart the stream by following the directions or returning to list view and clicking the dropdown option `restart` |

Reviewer 3: ```suggestion
| Suspended | Our system has disabled the stream because there is an issue and your logs could not be delivered. You may restart the stream by following the directions or returning to the list view and clicking the dropdown option `restart` |
```

> | Suspended | Our system has disabled the stream because there is an issue and your logs could not be delivered. You may restart the stream by following the directions or returning to list view and clicking the dropdown option `restart` |

Reviewer 3: The dashboard term used is `disabled`:

```suggestion
| Disabled | Our system has disabled the stream because there is an issue and your logs could not be delivered. You may restart the stream by following the directions or returning to list view and clicking the dropdown option `restart` |
```

## Streaming delivery errors, pauses, or suspensions. 

> Auth0 events are delivered to your server via a streaming mechanism that sends each event as it is triggered in our system. If a stream is paused or suspended, log deliveries will resume from the last delivery point upon stream resume or restart. Please note, however, that logs are only retained up to the retention period as noted on your account. Once the retention period is up, missing logs will not be available for retreival. 

Reviewer 3: `as it is triggered in our system` suggests that we attempt delivery in real-time and we make no such promises. We should reword it to avoid making that promise inadvertently

> Auth0 events are delivered to your server via a streaming mechanism that sends each event as it is triggered in our system. If a stream is paused or suspended, log deliveries will resume from the last delivery point upon stream resume or restart. Please note, however, that logs are only retained up to the retention period as noted on your account. Once the retention period is up, missing logs will not be available for retreival. 

Reviewer 3: ```suggestion
Auth0 events are delivered to your server via a streaming mechanism that sends each event as it is triggered in our system. If a stream is paused or disabled, log deliveries will resume from the last delivery point when the stream is resumed or restarted. Please note, however, that logs are only retained up to the retention period as noted on your account. Once the retention period is up, missing logs will not be available for retrieval. 
```

> Auth0 events are delivered to your server via a streaming mechanism that sends each event as it is triggered in our system. If a stream is paused or suspended, log deliveries will resume from the last delivery point upon stream resume or restart. Please note, however, that logs are only retained up to the retention period as noted on your account. Once the retention period is up, missing logs will not be available for retreival. 

Reviewer: ```suggestion
Auth0 events are delivered to your server via a mechanism that sends batches of log events as they are triggered in our system. If a stream is paused or suspended, log event deliveries will resume from the last delivery point once the stream is resumes or restarted. Please note that log events are only retained up to the retention period as noted on your account. Once the retention period is up, logs will not longer be available for retrieval. 
```

### Errors

> Errors (from the last 24 hours only) are populated in the Logs Health Table when logs could not be delivered. The log errors are populated with corresponding error codes and checkpoint information. Additional information on retrieving associated logs via checkpoint can be found here: https://auth0.com/docs/logs/guides/retrieve-logs-mgmt-api#get-logs-by-checkpoint. 

Reviewer 3: We may or may not want to tell them about this, because as to @machuga 's point, it is not possible to replicate the exact payload body using the startingCursor alone. More details here:

https://auth0.slack.com/archives/CEJHNE5QV/p1585847510013100

Let's remove this line. We might update the UI to remove the `copy to clipboard` action for the starting cursor. TBD

> Errors (from the last 24 hours only) are populated in the Logs Health Table when logs could not be delivered. The log errors are populated with corresponding error codes and checkpoint information. Additional information on retrieving associated logs via checkpoint can be found here: https://auth0.com/docs/logs/guides/retrieve-logs-mgmt-api#get-logs-by-checkpoint. 

Reviewer: ```suggestion
In the event that a batch of logs was unable to be delivered an errors will be logged in the Logs Health Table for up to 24 hours after the delivery attempt. The errors will include an error code as well as checkpoint information. Additional information on retrieving associated logs via checkpoint can be found here: https://auth0.com/docs/logs/guides/retrieve-logs-mgmt-api#get-logs-by-checkpoint. 
```

### Pause/Resume Streams

> Perhaps you need to pause streaming for a while during an upgrade or error resolution?

Reviewer: I am not sure we need this question.

> You may pause your stream by using the drop down option menu in the log stream list view and clicking 'pause.' Streams may be resumed by returning to the dropdown and clicking 'resume.'

Reviewer 3: ```suggestion
You may pause your stream by using the drop-down option menu in the log stream list view and clicking `Pause Stream'. Streams may be resumed by returning to the drop-down and clicking 'Resume Stream'.
```

> You may pause your stream by using the drop down option menu in the log stream list view and clicking 'pause.' Streams may be resumed by returning to the dropdown and clicking 'resume.'

Reviewer: ```suggestion
Streams can be paused by using the drop down option menu in the log stream list view and clicking 'pause.' Streams may be resumed by returning to the dropdown and clicking 'resume.'
```

All the docs are written in passive voice and I think this part should remain consistent with that. Also I am not sure if this should be documented, seems pretty basic. @tony-jones @erjohnson what do you think?

![Pausing a Stream](/media/articles/logs/streams-health/stream-pause.png "Pausing a Stream")

> If a stream is paused, you may resume it any time using the same steps. 

Reviewer 3: Is this line needed? Didn't we just explain how to resume a stream in the lines above?

Reviewer: agreed. not sure this whole section is needed TBH

### Suspensions

> If your server is unable to receive the stream, we will retry  to deliver the event; if the events cannot be received after several consecutive retries, we will suspend the stream. You can restart the stream by following the link in the suspension message or by returning to the list view and using the dropdown menu to click 'restart.' 

Reviewer 3: ```suggestion
If we receive an error while attempting to deliver events to your log stream, we will retry event delivery; if the events cannot be received after several consecutive retries, we will suspend the stream. You can restart the stream by following the link in the suspension message or by returning to the list view and using the drop-down menu to click `Restart Stream' 
```

> If your server is unable to receive the stream, we will retry  to deliver the event; if the events cannot be received after several consecutive retries, we will suspend the stream. You can restart the stream by following the link in the suspension message or by returning to the list view and using the dropdown menu to click 'restart.' 

Reviewer: ```suggestion
In the event that the stream target is unable to receive a delivery attempt, our system to retry several times; if the events cannot be successfully delivered after several consecutive retries, we will suspend the stream. The event can be restarted manually by following the link in the suspension message or by returning to the list view and using the dropdown menu to click 'restart.' 
```

> ### Resolving Errors and Undelivered Logs During Paused, Active, or Suspsended States

Reviewer 3: ```suggestion
### Resolving Errors during Paused, Active, or Suspended States
```

> Don't see logs on your end? Use the error table to find out why. We now log recent errors and their descriptions to help you investigate issues with your stream.

Reviewer: ```suggestion
The error table logs recent errors in the delivery of logs with a description description to help understand and troubleshoot issues with a stream.
```

![Disabled Stream](/media/articles/logs/streams-health/disabled-stream.png "A Disabled Stream")

1. On a stream's **Health** tab: please reference any errors and the associated counter information. 

> 2. Follow the instructions outlined here: https://auth0.com/docs/logs/guides/retrieve-logs-mgmt-api#get-logs-by-checkpoint

Reviewer 3: Again, we might remove this altogether.

Reviewer: I agree. not sure we need this section
